pkg/proxy/backend/authenticator.go

```diff
@@ -235,6 +231,18 @@ loop:
 			return err
 		}
 		if packetErr != nil {
+			// tiproxy pp enabled, tidb pp disabled, tls disabled => invalid sequence
+			// tiproxy pp disabled, tidb pp enabled, tls disabled => invalid sequence
+			for _, p := range []string{
+				"packets out of order",
+				"(nvali): d sequence",
+				"invalid sequence",
+				"PROXY Protocol",
+			} {
+				if strings.Contains(packetErr.Message, p) {
+					return errors.Wrap(ErrBackendPPV2, packetErr)
+				}
+			}
```

Comment:
I thought we could judge it by error ID. This error message may change anytime.

Reply:
If we can not judge the error ID, we can just report ErrBackendPPV2. I think in most cases it's ErrBackendPPV2 if reading the first packet fails.

Reply:
Done for 1156/8052, which may caused by other reasons, then return wrapped error.
If it contains "Proxy protocol" string, it returns unwrapped ErrBackendPPV2.

Reply:
Addressed comments. Reported the issue to TiDB.